analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.cs

```diff
@@ -172,5 +189,89 @@ public void Execute_PersistSymbols_InsideBlock()
             static ISymbol LocalReferenceOperationSymbol(IOperationWrapperSonar operation) =>
                 ((ILocalReferenceOperation)operation.Instance).Local;
         }
+
+        [TestMethod]
+        public void Execute_SimpleAssignmentWithLiteralOnLocalVariables_PropagatesConstraints()
+        {
+            var setter = new PreProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.Literal)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateCS(@"var a = true; Tag(""a"", a);", setter).Collector;
+            collector.Validate("Literal: true", x => x.State[x.Operation].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.Validate("SimpleAssignment: a = true (Implicit)", x => x.State[new IOperationWrapperSonar(((ISimpleAssignmentOperation)x.Operation.Instance).Target)].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
```

Comment:
Should be possible after addressing other comment about overload. If not, add this as well.
```suggestion
            collector.Validate("SimpleAssignment: a = true (Implicit)", x => x.State[((ISimpleAssignmentOperation)x.Operation.Instance).Target].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
```